app/controllers/payment-links/metadata/metadata-form.js

```diff
@@ -76,8 +95,17 @@ class MetadataForm {
   }
 }
 
-function isNotEmpty (value) {
+function isNotEmpty (value, existingMetadata) {
   return value && value.length !== 0
 }
 
+function isValidLength (value, existingMetadata) {
+  return value && value.length !== 0 && value.length <= META_DATA_MAX_LENGTH
+}
+
+function isNotDuplicate (value, existingMetadata) {
+  const found = existingMetadata.find(metadata => value === metadata['metadata-column-header'])
```

Comment:
Ideal to use an variable here to avoid typing mistakes and make this resilient to the ID changing in the future 
```suggestion
  const found = existingMetadata.find(metadata => value === metadata[fields.metadataKey.id])
```

Reply:
Done